src/ansys/fluent/core/solver/flobject.py

```diff
@@ -372,17 +388,49 @@ class String(SettingsBase[str], Textual):
     _state_type = str
 
 
-class Filename(SettingsBase[str], Textual):
+class InputOutputFileBase:
+    def is_input(self):
+        if self.file_purpose() == "input":
+            return True
+        return False
+
+    def before_execute(self, value):
```

Comment:
No before_execute in the base?

Reply:
As per our latest discussion, we have decided to use single base class (InputOutputFileBase) for Filename and FilenameList class instead of mixins.

Reply:
I updated the code now.